File: 15-03-18/java/03-Monk-and-Power-of-time.java
import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n=input.nextInt();
        Queue<Integer> que = new LinkedList<>();
        int ar[] = new int[n];
        
        for(int i=0;i<n;i++)
        {
            int tmp=input.nextInt();
            que.add(tmp);
        }
        
        for(int i=0;i<n;i++)
            ar[i]=input.nextInt();
        
        int ans=0;
        
        for(int i=0;i<n;i++)
        {
            ans++;
            while(que.peek()!=ar[i])
            {
                ans++;
                que.add(que.peek());
                que.remove();
            }
            que.remove();
        }
        System.out.println(ans);
        
        
    }
}

File: 15-03-18/c++/03-Monk-and-Power-of-time.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int n;
    cin>>n;
    int ar[n];
    queue<int> que;
    for(int i=0;i<n;i++)
    {
        int tmp;
        cin>>tmp;
        que.push(tmp);
    }
    for(int i=0;i<n;i++)
        cin>>ar[i];
    int ans=0;
    
    for(int i=0;i<n;i++)
    {
        ans++;
        while(que.front()!=ar[i])
        {
            que.push(que.front());
            que.pop();
            ans++;
        }
        que.pop();
    }
    cout<<ans<<endl;
}
